core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/dictionary/DictionaryMatcherOpExec.scala

```diff
@@ -0,0 +1,149 @@
+package edu.uci.ics.texera.workflow.operators.dictionary
+
+import edu.uci.ics.amber.engine.common.InputExhausted
+import edu.uci.ics.amber.engine.common.virtualidentity.LinkIdentity
+import edu.uci.ics.texera.workflow.common.operators.OperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{AttributeType, OperatorSchemaInfo}
+import org.apache.lucene.analysis.Analyzer
+import org.apache.lucene.analysis.en.EnglishAnalyzer
+import org.apache.lucene.analysis.tokenattributes.OffsetAttribute
+import scala.collection.mutable.Set
+import java.io.StringReader
+import scala.collection.Iterator
+import scala.collection.mutable.ListBuffer
+import scala.util.Either
+
+class DictionaryMatcherOpExec(
+    opDesc: DictionaryMatcherOpDesc,
+    operatorSchemaInfo: OperatorSchemaInfo
+) extends OperatorExecutor {
```

Comment:
Since this implementation is effectively a `map`, function, it's better to extend `MapOpExec`